src/ResourceManager/ApplicationInsights/Commands.ApplicationInsights/ApplicationInsights/SetApplicationInsightsDailyCap.cs

```diff
@@ -0,0 +1,134 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ApplicationInsights.Models;
+using Microsoft.Azure.Management.ApplicationInsights.Management.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ApplicationInsights
+{
+    [Cmdlet(VerbsCommon.Set, ApplicationInsightsDailyCapNounStr), OutputType(typeof(PSPricingPlan))]
+    public class SetApplicationInsightsDailyCapCommand : ApplicationInsightsBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Application Insights Component Object.")]
+        [ValidateNotNull]
+        public PSApplicationInsightsComponent ApplicationInsightsComponent { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Resource Id.")]
+        [ValidateNotNull]
+        public ResourceIdentifier ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Name.")]
+        [Alias(ApplicationInsightsComponentNameAlias, ComponentNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Daily Cap.")]        
+        public double? DailyCapGB { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = false,
+            HelpMessage = "Stop send notification when hit cap.")]
+        public SwitchParameter DisableNotificationWhenHitCap { get; set; }
+
+        [Parameter(
+            Position = 4,
+            Mandatory = false,
+            HelpMessage = "Enable send notification when hit cap.")]
+        public SwitchParameter EnableNotificationWhenHitCap { get; set; }
```

Comment:
@frankguodongchen these enable and disable switches should be in separate parameter sets so a user wouldn't be able to provide both of them in the same command